core/block/process/service.go

```diff
@@ -85,13 +97,33 @@ func (s *service) monitor(p Process) {
 			},
 		},
 	})
+	var updated chan struct{}
+	if updatableProcess, ok := p.(UpdatableProcess); ok {
+		updated = updatableProcess.Updated()
+	}
 	var prevInfo = info
 	for {
 		select {
+		case <-updated:
+			info := p.Info()
+			if !infoEquals(info, prevInfo) {
+				s.eventSender.BroadcastExceptSessions(&pb.Event{
+					Messages: []*pb.EventMessage{
+						{
```

Comment:
maybe we can avoid reading update in this loop at all? It will block the progress.Update call with some chance, even in case of buffered chan. Especially for the operations that call Update very frequently.

Instead you can leave the ticker only and update the info under the lock?

Reply:
I added new channel because sometimes `done` event can be send before the update event from ticker, so progress bar is finished with wrong information. So we can avoid such problems if during finish with assume, that totalCount == doneCount and put this information to `Info`